tests/test/org/commcare/test/utils/XmlComparator.java

```diff
@@ -0,0 +1,91 @@
+package org.commcare.test.utils;
+
+import org.kxml2.io.KXmlParser;
+import org.kxml2.kdom.Document;
+import org.kxml2.kdom.Element;
+import org.xmlpull.v1.XmlPullParser;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Hashtable;
+
+/**
+ * @author ctsims
+ */
+public class XmlComparator {
+
+    public static Document getDocumentFromStream(InputStream is) {
+        KXmlParser parser = new KXmlParser();
+
+        Document document = new Document();
+
+        try {
+            InputStreamReader reader = new InputStreamReader(is, "UTF-8");
+            parser.setInput(reader);
+            parser.setFeature(XmlPullParser.FEATURE_PROCESS_NAMESPACES, false);
+            document.parse(parser);
+            return document;
+        } catch(Exception e) {
+            throw new RuntimeException(e.getMessage());
+        }
+    }
+
+    public static void compareXmlDOMs(Document a, Document b) throws RuntimeException {
```

Comment:
Shouldn't this method return a boolean and be renamed to xmlDOMsEqual? Or you leave the name the same and return -1, 0, 1, making it a comparator